pkgs/development/beam-modules/default.nix

```diff
@@ -37,9 +37,23 @@ let
         buildMix = callPackage ./build-mix.nix {};
 
         # BEAM-based languages.
-        elixir = if versionAtLeast (lib.getVersion erlang) "18"
-                 then callPackage ../interpreters/elixir { debugInfo = true; }
-                 else throw "Elixir requires at least Erlang/OTP R18.";
+        elixir = elixir-1_4;
+
+        elixir-1_5 = lib.callElixir ../interpreters/elixir/1.5.nix "18" {
+                       inherit rebar erlang;
+                       debugInfo = true;
+                     };
+
+        elixir-1_4 = lib.callElixir ../interpreters/elixir/1.4.nix "18" {
```

Comment:
I think it makes more sense to add the otp version as an optional argument for the elixir builder instead, then it could be overridden by the specific elixir expression if needed.